script/dropbox.js

```diff
@@ -1,362 +1,384 @@
+                                        text: _('load from slot') + n + ' ' + DropboxConnector.prepareSaveDate(savegame.get('timestamp')),
+                                        nextScene: 'end',
+                                        onChoose: function() {
+                                            DropboxConnector.log('Load from slot ' + n + ' initiated');
+                                            // timeout prevents error due to fade out animation of the previous event
+                                            Engine.setTimeout(function() {
+                                                DropboxConnector.log('Load from slot ' + n);
+                                                DropboxConnector.loadGameFromDropbox(n);
+                                            }, 1000);
+                                        }
+                                    };
+                                }
+                            });
+
+                            buttons.cancel = {
+                                text: _('cancel'),
+                                nextScene: 'end'
+                            };
+
+                            return buttons;
+                        }())
+                    }
+                }
+            });
+        },
+
+        savedtoDropboxEvent: function(success) {
+            Events.startEvent({
+                title: _('Dropbox Export / Import'),
+                scenes: {
+                    start: {
+                        text: success ? [_('successfully saved to dropbox datastorage')] : [_('error while saving to dropbox datastorage')],
+                        buttons: {
+                            'ok': {
+                                text: _('ok'),
+                                nextScene: 'end'
+                            }
+                        }
+                    }
+                }
+            });
+        },
+
+        /**
+         * ***************
+         * functional code
+         * ***************
+         */
+
+        /**
+         * Initiate dropbox connection
+         *
+         * @param interactive
+         * @param callback
+         */
+        connectToDropbox: function(interactive, callback) {
+
+            DropboxConnector.log('start dropbox');
+
+            var client = this.client;
+
+            client.authenticate({
+                interactive: interactive
+            }, function(error) {
+                if (error) {
+                    DropboxConnector.log('Dropbox Authentication error: ' + error);
+                }
+            });
 
-    /**
-     * ***************
-     * functional code
-     * ***************
-     */
+            if (client.isAuthenticated()) {
 
-    /**
-     * Initiate dropbox connection
-     *
-     * @param interactive
-     * @param callback
-     */
-    connectToDropbox: function (interactive, callback) {
+                var datastoreManager = client.getDatastoreManager();
```

Comment:
Someone else sign off on this code please, I haven't played with Dropbox stuff myself too much